File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        nums.sort()

        def bt(index, curr_set):
            if sum(curr_set) == target:
                res.append(curr_set[:])
                return
            for i in range(index, len(nums)):
                curr_set.append(nums[i])
                if sum(curr_set) < target:
                    bt(i, curr_set)
                else:
                    bt(i+1, curr_set)
                curr_set.pop()
        bt(0, [])
        return res

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int pre = 1;
        int[] res = new int[nums.length];

        for(int i=0; i<nums.length; i++){
            res[i] = pre;
            pre *= nums[i];
        }
        
        int post = 1;
        for(int j=nums.length-1; j>=0; j--){
            res[j] = res[j] * post;
            post *= nums[j]; 
        }
        return res;
    }
}  
